docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -38,6 +38,9 @@ When compiling code in any of the situations in the below table, add the corresp
 | for x86_64           | `-fcf-protection=full`         |
 | for aarch64          | `-mbranch-protection=standard` |
 | for production code  | `-fno-delete-null-pointer-checks -fno-strict-overflow -fno-strict-aliasing -ftrivial-auto-var-init=zero` |
+| for project developers | `-std=gnu2x -Werror=implicit-function-declaration -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=old-style-definition -Werror=implicit-int` |
```

Comment:
Recommendations on language dialects go beyond how I understand the scope of the guide. IMHO that is better left up to individual projects, which may have other considerations, such as portability, that go beyond security aspects.

`gnu2x` in particular is not a good recommendation as [that name has been deprecated in favor `gnu23` in GCC](https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html) (and would not work for C++ code either).

Reply:
The reason for `gnu2x` is documented in the proposed text, it's more portable.

HOWEVER: I incorrectly remembered *why* `-std` was being proposed by Weimer. I think this argument makes sense, but it should be explained. I've added that rationale to this proposed text.